### ISSUE TITLE
[A11y] Playlist edit page

### ISSUE THREAD

**Issue by Dananji** [avalonmediasystem/avalon#6354]
### Description

From the accessibility checks performed on playlist related pages in https://github.com/avalonmediasystem/avalon/issues/6312, the following were identified specifically for playlist edit page.
| Success criterion | WCAG 2.1 Level | Accessibility checker | Error | Relevant HTML element |
| --- | --- | --- | --- | --- |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE| Button missing a text alternative | Playlist item edit button with SVG |
| 1.4.3 Contrast | Level AA | SiteImprove & WAVE | Color contrast (3.95:1) doesn't meet min requirement | `No description` and `No tags` text in in playlist details |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<input>` form field for playlist item position | 
| 3.3.2 Labels or Instructions | Level A | WAVE & axeDevtools | Ensure an element's role supports its ARIA attributes | Edit playlist details button and cancel edits button |

The input form related issues are handled in #6353

For the ARIA attributes issue, more information was provided by `axeDevtools`: https://dequeuniversity.com/rules/axe/4.10/aria-allowed-attr?application=AxeFirefox

### Done Looks Like
 - [ ] Add a screen reader accessible label to the playlist details edit button
 - [ ] Fix color contrast for greyed out text in the details panel
 - [ ] Add labels for the input form fields for playlist item positions
 - [ ] Both playlist item details edit button and cancel buttons have ARIA attributes supported by the element's role
